### PR TITLE
Update traefik ingress docs

### LINKED ISSUE #1633: Update Linkerd Ingress Traffic doc
From Traefik v2.10 it supports sending requests directly to service endpoint. Github issue https://github.com/traefik/traefik/pull/9740. 

You might want to to update your Ingress traffic doc https://linkerd.io/2.13/tasks/using-ingress/ to reflect the same.

### PR DESCRIPTION
**Pull request by chicocvenancio**
Fixes #1633

### CODE REVIEW

File: linkerd.io/content/2.16/tasks/using-ingress.md [hunk 1]
```diff
@@ -77,8 +77,8 @@ Common ingress options that Linkerd has been used with include:
 - [Nginx (community version)](#nginx-community-version)
 - [Nginx (F5 NGINX version)](#nginx-f5-nginx-version)
 - [Traefik](#traefik)
-  - [Traefik 1.x](#traefik-1x)
-  - [Traefik 2.x](#traefik-2x)
+  - [Traefik with normal mode](#traefik-normal-mode)
+  - [Traefik with ingress mode](#traefik-ingress-mode)
```
Comment (kflynn) [inline linkerd.io/content/2.16/tasks/using-ingress.md:81]: Can you make clear even here that any version prior to 2.10 really must use ingress mode? 🙂

File: linkerd.io/content/2.16/tasks/using-ingress.md [hunk 2]
```diff
@@ -207,89 +207,56 @@ spec:
...
+      nativeLB: true
 ```
 
-The important annotation here is:
+### Traefik with ingress mode {#traefik-ingress-mode}
 
-```yaml
-ingress.kubernetes.io/custom-request-headers: l5d-dst-override:web-svc.emojivoto.svc.cluster.local:80
-```
+Previous versions of Ttraefik needed to use [ingress mode](#ingress-mode),
```
Comment (kflynn) [inline linkerd.io/content/2.16/tasks/using-ingress.md:245]:
```suggestion
Versions of Traefik prior to 2.10 must use [ingress mode](#ingress-mode),
```

File: linkerd.io/content/2.16/tasks/using-ingress.md [hunk 3]
```diff
@@ -207,89 +207,56 @@ spec:
...
-Definition to add the `l5d-dst-override` header.
-
-The YAML below uses the Traefik CRDs to produce the same results for the
-`emojivoto` application, as described above.
+The YAML below uses the Traefik custom resources to configure a route and
+headers for the `emojivoto` application.
 
 ```yaml
-apiVersion: traefik.containo.us/v1alpha1
+apiVersion: traefik.io/v1alpha1
```
Comment (kflynn) [inline linkerd.io/content/2.16/tasks/using-ingress.md:259]: What version of Traefik is required for this API group change?

**Review by kflynn — CHANGES_REQUESTED** [review]
Thanks @chicocvenancio! There are a few tweaks that I'd like to see, but this looks way better than what we had before and I appreciate it! 🙂

File: linkerd.io/content/2.16/tasks/using-ingress.md (hunk 3 above)
Comment (chicocvenancio) [reply linkerd.io/content/2.16/tasks/using-ingress.md:259]: Version 2.10 as well. `traefik.containo.us` is dropped in v3.

File: linkerd.io/content/2.16/tasks/using-ingress.md [hunk 4]
```diff
@@ -207,86 +207,53 @@ spec:
...
-### Traefik 1.x {#traefik-1x}
+With traefik versions 2.10 and newer "Kubernetes Service Native Load-Balancing"
+can be set in the Custom Resource called [`IngressRoute`](
+  https://docs.traefik.io/providers/kubernetes-crd/) with the
+`services[n].nativeLB` field.
 
-The simplest way to use Traefik 1.x as an ingress for Linkerd is to configure a
-Kubernetes `Ingress` resource with the
-`ingress.kubernetes.io/custom-request-headers` like this:
+The YAML bellow exemplifies an IngressRoute for `emojivoto` application.
```
Comment (kflynn) [inline linkerd.io/content/2.16/tasks/using-ingress.md:221]:
```suggestion
The YAML below exemplifies an IngressRoute for `emojivoto` application.
```

**Review by kflynn — CHANGES_REQUESTED** [review]
Thanks @chicocvenancio! This looks great save for one small typo.

**Comment by kflynn** [comment]
Also, it would be lovely if you could add the [DCO](https://developercertificate.org/) signature to your commits (or just confirm here that you're OK with the DCO).

File: linkerd.io/content/2.16/tasks/using-ingress.md (hunk 3 above)
Comment (chicocvenancio) [reply linkerd.io/content/2.16/tasks/using-ingress.md]: I left new api group for versions 2.10 and up example but kept old group for the legacy example.

**Comment by chicocvenancio** [comment]
Rebased to sign off all commits and fixed the typo.

**Review by kflynn — APPROVED** [review]
Many thanks!! 🙂

**Comment by kflynn** [comment]
Looks like this needs to have `main` merged before we can land it? Sorry for the extra hassle!

**Comment by kflynn** [comment]
Thanks for sticking with this one, @chicocvenancio! 🙂

**Pull request merged**